packages/react-magma-dom/src/components/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,203 @@
+import * as React from 'react';
+import styled from '../../theme/styled';
+import { XOR } from '../../utils';
+import { css } from '@emotion/core';
+import { transparentize } from 'polished';
+
+import { ThemeContext } from '../../theme/ThemeContext';
+import { ThemeInterface } from '../../theme/magma';
+import { Button, ButtonProps, ButtonSize } from '../Button';
+import { ButtonColor } from '../Button';
+import { IconButton } from '../IconButton';
+import { IconProps } from 'react-magma-icons';
+import { ToggleButtonGroupContext } from '../ToggleButtonGroup/ToggleButtonGroup';
+
+export interface ToggleButtonTextProps extends ButtonProps {
+  /**
+   * Sets a disabled state for a button.
+   */
+  disabled?: boolean;
+  /**
+   * @internal
+   */
+  isChecked?: boolean;
+  isInverse?: boolean;
+  /**
+   * Changes the button size: 'small', 'medium', and 'large'.
+   */
+  size?: ButtonSize;
+  /**
+   * @internal
+   */
+  theme?: ThemeInterface;
+}
+
+export interface ToggleButtonIconProps extends ButtonProps {
+  /**
+   * Sets the aria-label which is required for icon only buttons.
+   */
+  'aria-label'?: string;
+  /**
+   * Sets a disabled state for a button.
+   */
+  disabled?: boolean;
+  hasLabel?: boolean;
```

Comment:
Missing prop description

Reply:
This should be internal only I believe?

Reply:
If we don't expect users to use it, then it shouldn't be a prop and it should just be a variable within the component.